### PR TITLE
Add a lua method for querying all actors that are currently InWorld

### PR DESCRIPTION
**Pull request by abcdefg30**
As requested to fight the abuse of `Map.ActorsInBox`. Closes https://github.com/OpenRA/OpenRA/issues/8549.

### CODE REVIEW

**Comment by abcdefg30** [comment]
I'll updated all our maps to use this instead of `Map.ActorsInBox` tomorrow, to justify the `For Stable`.

**Comment by abcdefg30** [comment]
Updaded. I can squash the `Replace ...` commits if wished.

**Comment by obrakmann** [comment]
Looks good to me, :+1:

File: OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs [hunk 1]
```diff
@@ -54,6 +55,20 @@ public bool All(LuaValue[] collection, LuaFunction func)
 			return true;
 		}
 
+		[Desc("Returns the original collection filtered with the func.")]
+		public LuaTable Where(LuaValue[] collection, LuaFunction func)
+		{
+			var t = Context.CreateTable();
+
+			foreach (var c in collection)
+				using (var ret = func.Call(c))
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs]: This call should be disposed.

```diff
diff --git a/OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs b/OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs
index 50ca3e5..1172b6a 100644
--- a/OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs
+++ b/OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs
@@ -37,9 +37,13 @@ public bool Any(LuaValue[] collection, LuaFunction func)
 			foreach (var c in collection)
 				using (var ret = func.Call(c))
 				using (var result = ret.FirstOrDefault())
+				{
 					if (result != null && result.ToBoolean())
 						return true;
 
+					ret.Dispose();
+				}
+
 			return false;
 		}

```

Comment (abcdefg30) [inline OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs]: Um, that doesn't look like the `Where` function, but rather like `Any`...
Anyway, I could do that here in theory, but should I? I mean we don't do this anywhere here.

Comment (Mailaender) [inline OpenRA.Mods.Common/Scripting/Global/UtilsGlobal.cs]: > The using statement ensures that Dispose is called even if an exception occurs while you are calling methods on the object.

https://msdn.microsoft.com/en-us/library/yh598w02.aspx?f=255&MSPPError=-2147217396

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/211bbe61b097ead3fd0532c88219ceae1b7fdfa7...261708038bc091af20c43928628338e3b54098fd)

**Pull request merged**